bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/zk/ZKMetadataDriverBase.java

```diff
@@ -247,6 +256,19 @@ protected void initialize(AbstractConfiguration<?> conf,
             acls);
     }
 
+    private synchronized void handleState(Watcher.Event.KeeperState zkClientState) {
```

Comment:
this is `synchronized` so we must add a 
synchronized boolean isMetadataServiceAvailable()

methodm otherwise we are not accessing the field properly